sdks/java/core/src/main/java/org/apache/beam/sdk/util/Rebundle.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.util;
+
+import java.util.concurrent.ThreadLocalRandom;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.Values;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+
+/**
+ * A {@link PTransform} that returns a {@link PCollection} equivalent to its input, but prevents
+ * fusion of the surrounding transforms, so that the runner will split the returned {@link
+ * PCollection} into bundles for parallel processing independently.
+ */
+public abstract class Rebundle {
+  /**
+   * Creates a {@link Rebundle} transform that produces a collection that the runner can split into
+   * bundles in an arbitrarily fashion.
+   */
+  public static <T> Default<T> create() {
```

Comment:
Don't understand this, can you elaborate what you mean by "allowing all values to pass through the state machine for each key"? With unkeyed rebundle, there's no keys at all, so no per-key state either, right?